library/aes.c

```diff
@@ -1336,12 +1336,7 @@ int mbedtls_aes_self_test( int verbose )
             aes_tests = aes_test_ecb_enc[u];
         }
 
-        /*
-         * AES-192 is an optional feature that may be unavailable when
-         * there is an alternative underlying implementation i.e. when
-         * MBEDTLS_AES_ALT is defined.
-         */
-        if( ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE && keybits == 192 )
+        if( ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE )
```

Comment:
If a selftest should be handled in the alternative implementation as well, most of the time it will end up with copy \ pasting code.
Perhaps we should have some hooks for selftest callbacks, as suggested in https://github.com/ARMmbed/mbedtls/pull/1598#discussion_r187421900 but I am not sure it is part of the scope of this PR

Reply:
Perhaps we should just remove the changes in the self test in this PR, since they will be removed when we add support for  an alternative self test?

Reply:
Yes, I'm recommending we at least "remove the changes in the self test in this PR, since they will be removed when we add support for an alternative self test". Additionally, it shouldn't be too hard to make a new commit to make self-tests ALT-implementable in this PR. We can add hooks later so folks can avoid copy paste, but that needs further design work and is out of scope for this PR; a simple change would allow the self-tests to be ALT-implementable.

Reply:
>it shouldn't be too hard to make a new commit to make self-tests ALT-implementable in this PR

Right, it's just maintenance, having an additional self-test_ALT for every module. Having one self-test_ALT is also not recommended, since most of the time, only a few self tests will actually require alternative implementation

Reply:
I agree that self tests should pass, but different hardware accept different inputs, for example different buffer alignments.
I discussed with @Patater offline, and we will keep the current self test changes, accepting the `FEATURE_UNAVALILABLE`. (see [comment](https://github.com/ARMmbed/mbedtls/pull/1716#issuecomment-408849381))